django/db/models/fields/related.py

```diff
@@ -1447,8 +1447,10 @@ def _check_table_uniqueness(self, **kwargs):
             if model != self.remote_field.through and model._meta.managed
         }
         m2m_db_table = self.m2m_db_table()
-        if m2m_db_table in registered_tables:
-            model = registered_tables[m2m_db_table]
+        model = registered_tables.get(m2m_db_table, False)
```

Comment:
Any reason to pass `False` here? The default behaviour of `get()` returning `None` seems to be enough.